### PR TITLE
Update to [PyTrack 0.1.1]

### LINKED ISSUE #51: Support numpy arrays in `DVC.results()` and `DVC.params()`
Numpy Arrays are very common results and potentially parameters.
Currently they have to be converted with `np.ndarry.tolist()` manually.
This could be automated.

*Possible implementations*
- convert them to lists
- allow custom result types, that use e.g., np.save and np.load instead of json files.

### LINKED ISSUE #57: Allow named stages
Give the possibilty to name a stage, e.g.,

```py
from pytrack import PyTrack, DVC

@PyTrack(name="WriteInputToOutput")
class HelloWorld:
    def __init__(self):
        self.input = DVC.params()
        self.output = DVC.result()
    
    def __call__(self, input):
        self.input = input
    def run(self):
        self.output = self.input
```

### LINKED ISSUE #58: Issue with running from `__main__`
The following code fails:
```py
from pytrack import PyTrack, DVC, PyTrackProject


@PyTrack(package=False)
class HelloWorld:
    def __init__(self) -> None:
        self.output = DVC.result

    def __call__(self, *args, **kwds):
        pass

    def run(self):
        self.output = "Hello World"


if __name__ == "__main__":
    project = PyTrackProject()
    project.create_dvc_repository()

    hello_world = HelloWorld()
    hello_world()
```

with 

`
ImportError: cannot import name 'HelloWorld' from '__main__' (unknown location)
ERROR: failed to reproduce 'dvc.yaml': failed to run: python -c "from __main__ import HelloWorld; HelloWorld(id_=0).run()", exited with 1
`

### LINKED ISSUE #65: Loaded stage prohibts `__call__` on non-loaded stage
To reproduce:
```py
from pytrack import PyTrack, DVC

@PyTrack()
class ComputeA:
    """PyTrack stage A"""

    def __init__(self):
        self.inp = DVC.params()
        self.out = DVC.result()

    def __call__(self, inp):
        self.inp = inp

    def run(self):
        self.out = np.power(2, self.inp).item()


a = ComputeA()
b = ComputeA(id_=0)
a(3)
```

This raises `ValueError: This stage is being loaded. Parameters can not be set!`

### PR DESCRIPTION
**Pull request by PythonFZ**
# Changelog
- make all `PyTrack` classes callable, even if they don't define a specific call method (see #56 )
- fix #58 
- fix #57 
- fix #51 (only numpy, pandas is not supported yet)
- fix #65 
- add `DVC.metrics_no_cache`
- `DVC.params/deps` no longer converts a string to a Pathlib object by default, but rather stores the type if defined in the `serializer.py` as a dictionary. So passing a `DVC.deps(Path(str))` will yield a `Path` and `DVC.deps(str)` will yield a `str`. This only applies if no entry for the path is defined in `DVCParams`

For #63 an API is proposed as follows:
```py
from pytrack import PyTrack, DVC
from pytrack.core.parameter import PyTrackOption
import pandas as pd

class DVC_pd(PyTrackOption):
    def _get(self, instance, owner):
        name = self.get_name(instance)
        return pd.read_csv(f"{name}.csv")
    def _set(self, instance, value):
        name = self.get_name(instance)
        value.to_csv(f"{name}.csv")

@PyTrack(nb_name='CustomDVC.ipynb')
class HelloWorld:
    def __init__(self):
        self.df = DVC_pd()
        self.df_out = DVC.outs('df.csv')   # in the current version this is mandatory, so DVC knows about the file!
    def __call__(self, df):
        self.df = df
    def run(self):
        print('HelloWorld')
```

### CODE REVIEW

File: pytrack/core/decorator.py [hunk 1]
```diff
@@ -25,20 +26,29 @@
 
 
 class PyTrack:
```
Comment (SamTov) [inline pytrack/core/decorator.py:28]: class docstring

File: pytrack/core/decorator.py [hunk 2]
```diff
@@ -129,35 +139,54 @@ def apply_decorator(self):
...
 
         def wrapper(cls: TypeHintParent, *args, id_=None, **kwargs):
             log.debug(f"Got id_: {id_}")
-            pytrack_parent = PyTrackParent(child=cls)
-            pytrack_parent.pre_init(id_=id_)
 
-            setattr(type(cls), "pytrack", property(lambda self_: pytrack_parent))
-            # setattr(cls, "pytrack", pytrack_parent)
-            # cls.pytrack = pytrack_parent
+            def map_pytrack_to_dict(self_):
```
Comment (SamTov) [inline pytrack/core/decorator.py:163]: what is the self_?

**Review by SamTov — CHANGES_REQUESTED** [review]
Please address comments.

File: pytrack/core/decorator.py (hunk 2 above)
Comment (PythonFZ) [reply pytrack/core/decorator.py:166]: fixed in 2382977

File: pytrack/core/decorator.py (hunk 1 above)
Comment (PythonFZ) [reply pytrack/core/decorator.py:28]: fixed in 2382977

**Review by SamTov — APPROVED** [review]
Thanks.

**Pull request merged**